src/datamodel/game_manager.rs

```diff
@@ -0,0 +1,31 @@
+use crate::datamodel::game_status::GameStatus;
+use crate::datamodel::player_action_manager::PlayerActionManager;
+use crate::datamodel::turn_manager::TurnManager;
+use crate::datamodel::Game;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+pub struct GameManager<'a> {
+    pub game: Game,
+    pub status: GameStatus<'a>,
+    pub turn_manager: Rc<TurnManager>,
```

Comment:
Idem here. Couldn't we replace this for a mutable reference? I'm just guessing.. Frankly I don't think I have the knowledge to advice properly in this matter. However, I'd try to avoid to complex implementation.